simulator/modules/fetch/bpu/bpu.h

```diff
@@ -36,12 +36,13 @@ class BaseBP
     BaseBP() = default;
 public:
     virtual bool is_taken( Addr PC) const = 0;
+    virtual bool is_hit( Addr PC) const = 0;
     virtual Addr get_target( Addr PC) const = 0;
     virtual void update( const BPInterface& bp_upd) = 0;
 
     BPInterface get_bp_info( Addr PC) const
     {
-        return BPInterface( PC, is_taken( PC), get_target( PC)); 
+        return BPInterface( PC, is_taken( PC), get_target( PC), is_hit( PC));
```

Comment:
The buzzy thing here is that cache looked up thrice here, and it is not good from simulation speed point.
Can we do that instead:
```c++
if ( is_hit(PC))
    return BPInterface( PC, is_taken( PC), get_target( PC), true);

return BPInterface( PC, false, PC + 4, false);
```
?

Anyway, I plan to complete separation of BTB and BP and to re-write that place...